app/assets/javascripts/discourse/app/lib/register-topic-footer-button.js

```diff
@@ -51,6 +51,12 @@ export function registerTopicFooterButton(button) {
 
     // display order, higher comes first
     priority: 0,
+
+    // is this button displayed for logged-in users ?
+    displayForUser: true,
+
+    // is this button displayed for anonymous users ?
+    displayForAnonymous: false,
```

Comment:
I think these options are not ideal and slightly too verbose because if you want to display for an anonymous user but not a logged in user, you would have to pass in `displayForUser: false` and `displayForAnonymous: true`. Perhaps we want look into adding support for a `shouldDisplay` callback function which would have access to the `currentUser` and the `topic` attributes?